### ISSUE TITLE
"type.is is not a function" when using constructor function with assert.union

### ISSUE THREAD

**Issue by ivan** [gcanti/flowcheck#21]
The `x instanceof type` check in `validate` works when passing in a constructor function, but not when using an `assert.union` that contains a constructor function.

```js
# iojs
> assert = require('./assert');
> assert.check(3, assert.number)
3
> assert.check(new Date(), Date)
Thu Jun 25 2015 03:14:21 GMT+0000 (UTC)
> assert.check(new Date(), assert.union([assert.number, Date]))
TypeError: type.is is not a function
    at /home/at/flowcheck/assert.js:225:19
    at Array.some (native)
    at Type.validate (/home/at/flowcheck/assert.js:224:15)
    at validate (/home/at/flowcheck/assert.js:77:17)
    at Object.check (/home/at/flowcheck/assert.js:279:16)
    at repl:1:8
    at REPLServer.defaultEval (repl.js:154:27)
    at bound (domain.js:254:14)
    at REPLServer.runBound [as eval] (domain.js:267:12)
    at REPLServer.<anonymous> (repl.js:308:12)
```

**Comment by ivan** [comment]
`list` doesn't have this problem, so maybe `union` should call `validate` like `list` does.

**Comment by gcanti** [comment]
Thanks @ivan, good catch. Just published v0.2.7 with the fix

**Comment by ivan** [comment]
Thank you!
